Anonymous events (#9078)
PostHog/posthog.com: contents/blog/analytics-pricing.md

---
date: 2024-07-31
title: "We’ve decided to make less money [Part 2]"
featuredImage: >-
  https://res.cloudinary.com/dmukukwp6/image/upload/posthog_is_the_cheapest_e77c4ea4a5.jpg
author:
  - andy-vandervell
featuredImageType: full
category: PostHog news
---

## TL;DR

- Last week, we [slashed the cost of session replay](/blog/session-replay-pricing) by more than 50%, making PostHog the cheapest session replay tool in its class.

- Today we're announcing Part 2, a change to our pricing that makes tracking events by anonymous users up to 80% cheaper.

- We'll see an ~30% decline in product analytics revenue from the change, but it will unlock several new use cases, such as web analytics at massive scale, and tracking high-volume backend events, that were previously prohibitively expensive.

## What's changing?

Up to now, we charged the same for all events we ingested. This is standard practice among product analytics tools. Popular tools like Mixpanel, Amplitude, and Heap all charge customers the same for each tracked user, session, or event, regardless of the cost to them.

But now we’re changing that by splitting our product analytics pricing into two event types:

> - **Identified events.** Events generated by logged-in users associated with a person profile, which include data like a user's name and email address. Identified events let you create user-specific insights and cohorts based on user properties. They cost the same as our existing event pricing.

Author: ```suggestion
- **Identified events.** Events generated by logged-in users associated with a person profile, which can include data like a user's name and email address. Identified events let you create user-specific insights and cohorts based on person properties. They cost the same as our existing event pricing.
```

> - **Anonymous events.** Events that only include non-identifying data, such as referral data (UTMs, domains, etc), device type, and basic location information. Anonymous events are between 55% and 80% cheaper for most of our customers, and will soon become the default event type.

Reviewer: ```suggestion
- **Anonymous events** only include non-identifying data, such as referral data (UTMs, domains, etc), device type, and basic location information. Anonymous events are between 55% and 80% cheaper for most of our customers, and will soon become the default event type.
```

![event types](https://res.cloudinary.com/dmukukwp6/image/upload/events_explained_0bccb88356.png)

> This means a website owner who sends 10m events per month would pay just $324 per month if they switched to only sending anonymous events, saving $9,072 per year in the process.

Reviewer: This doesn't really have a context and should probably go below the table. 
```suggestion
```

You can estimate your potential savings using the calculator on [our pricing page](/pricing), but here's a breakdown of how identified and anonymous events compare:

> | **Monthly events** | **Identified events** | **Anonymous events** | **Change** |

Reviewer: The green text is a nice touch!

Author: Thank Cory for that

| <strong class="text-15px">50-100 million</strong>   | <span class="text-[15px] font-semibold">$0.0000364</span><span class="text-sm opacity-70">/event</span>  | <span class="text-[15px] font-semibold">$0.0000218</span><span class="text-sm opacity-70">/event</span> | <strong class="text-green">40% cheaper</strong> |
| <strong class="text-15px">100-250 million</strong>  | <span class="text-[15px] font-semibold">$0.0000187</span><span class="text-sm opacity-70">/event</span>  | <span class="text-[15px] font-semibold">$0.0000150</span><span class="text-sm opacity-70">/event</span> | <strong class="text-green">20% cheaper</strong> |
| <strong class="text-15px">250+ million</strong>     | <span class="text-[15px] font-semibold">$0.0000100</span><span class="text-sm opacity-70">/event</span>  | <span class="text-[15px] font-semibold">$0.0000090</span><span class="text-sm opacity-70">/event</span> | <strong class="text-green">10% cheaper</strong> |

> 

Reviewer: ```suggestion

This means a website owner who sends 10m events per month would pay just $324 per month if they switched to only sending anonymous events, saving $9,072 per year in the process.

```

> ## When should I use anonymous events?

Reviewer: ```suggestion
## When should I use anonymous events?

```

Anonymous events are ideal for tracking users who aren't logged in, or for use cases where you only need to analyze behavior at an aggregate level, such as:

- Visitors to your marketing website

> - Readers of a news website

Reviewer: This one feels a bit weak and really non-ICP
```suggestion
```

- Shoppers on an e-commerce store
- Users of a B2C mobile app
- Tracking API and backend events
- Events generated by LLMs
- Error monitoring

If you mostly track logged-in users of a B2B product, however, you should continue to do so. 

> We're supporting both because we want people to have the level of analysis they need for their use case at a fair price, rather than forcing them to pay more when they only need basic data on individual users.

Reviewer: This sentence is long and confusing and we explain it better elsewhere (earlier)
```suggestion
```

## Why are anonymous events so much cheaper?

The ELI5 version is that they cost less to ingest and process, so we're passing that saving onto you.

Here's the slightly longer version:

> - When you call `posthog.identify`, or an identified user performs an event, we do a bunch of processing to create, update, or merge person profiles in our backend. This is one of the most complex steps in [our ingestion pipeline](/docs/how-posthog-works/ingestion-pipeline), which makes it expensive.

Reviewer: ```suggestion
- When you call `posthog.identify` or capture an event from an identified user, we do a bunch of processing to create, update, or merge person profiles in our backend. These are stored separately from events in Postgres, making this is one of the most complex and expensive steps in [our ingestion pipeline](/docs/how-posthog-works/ingestion-pipeline).
```

> - When ingesting anonymous events, we store data in a sessions table that captures session properties, such as the referring domain and the URLs visited. This table is much cheaper for us to update and query. It's faster, too.  

Reviewer: ```suggestion
- When ingesting anonymous events, we can entirely rely on our cheaper, faster ClickHouse tables. For example, we use a sessions table to capture session properties, such as the referring domain and the URLs visited.
```

## Why are you deliberately making less money?

Because it's the right thing to do for both our users and our business.

Most of our competitors are inefficient. They employ huge [outbound sales teams](/founders/negotiate-software-better) to grow revenue. Their salaries and commissions for closing deals are passed onto their customers through higher prices.

In contrast, we're 100% inbound, we [grow mostly through word of mouth](/handbook/how-we-get-users), and we charge based on actual usage. 

We don't believe in loss-leaders, so we make a modest positive margin on each event sold, but what we charge is directly connected to what it costs us, not what we think we can get away with charging.

We grow our revenue through helping you grow, and onboarding you onto other tools, like session replay, feature flags, and surveys.

We think this is better than trying to squeeze you for every cent you have. You have a great experience, and we'll enjoy better retention and word of mouth for doing the right thing.

> ## FAQ
>
> ### Do I still get 1 million free events?

> > es, your first 1 million events each month are free, regardless of what type of events you're sending.

Reviewer: ```suggestion
> Yes, your first 1 million events each month are free, regardless of what type of events you're sending.
```

>
> ### What are the limitations of anonymous events?
>

> > When sending anonymous events, you cannot:

Reviewer: It's not really "when" because you could capture anonymous events now but still have identified events.
```suggestion
> With anonymous events, you cannot:
```

> - Filter on persons (e.g. an individual user).
> - Do multi-touch attribution – only last touch is supported.
> - Target by person properties for feature flags, A/B tests, and surveys.
> - Create cohorts filtered on person properties (e.g. paying vs non-paying users).
> - View a person’s profile in the app, or query the persons table in our SQL insights.
> - Use [groups analytics](/docs/product-analytics/group-analytics) on anonymous events.
>
> If you need the above functionality for all your traffic, you don't need to change anything. Continue to send person data with your events.
>
> ### Can I use anonymous and identified events together on the same project?
>

> > Absolutely. You can send identified events where you need more detailed information, and skip them when you don't need that detail.

Reviewer: Skipping makes it sound like dropping
```suggestion
> Absolutely. You can capture identified events where you need more detailed information and anonymous events when you don't need that detail.
```

>
> Here's how we use them:
>
> - On our website, most users are tracked using anonymous events. When people visit our website, referral data like Entry UTM Source are stored in the sessions table.
>
> - When someone signs up to PostHog and becomes an identified user, referral data like `Entry UTM Source` are merged into the identified user's person profile as `Initial UTM Source`, `Initial Referring Domain`, and so on.
>
> - We run our website and product on the same project. This is how we recommend using PostHog for tracking behavior on both your product and website.
>
> Note, though, that once a user becomes identified, all events they perform are identified events, because we need to check and update their profile each time.
> 
> ###  How do I start using anonymous events?
>

> > For most users, it only requires a simple config change to include `process_persons: "identified_only" in the initialization of your JavaScript Web SDK or snippet.

Reviewer: ```suggestion
> For most users, it only requires a simple config change to include `process_persons: "identified_only"` in the initialization of your JavaScript Web SDK or snippet.
```

> 
> This will track anonymous events until an identifying action is taken – i.e. using `identify()`, `group()`, setting person properties with `$set`, etc.
>
> You can get more details on this and other use cases, like server-side events, in our [persons documentation](/docs/data/persons).
>
> ### I have a question you haven't answered...
>
> It may be covered in our [person](/docs/data/persons) and [person properties](/docs/product-analytics/person-properties) docs.
>

> > If not, ask it below, in [our community forum](/questions), or raise a support ticket in the app.

Reviewer: Why is this all a quote?

---
date: 2024-08-01
title: "We’ve decided to make less money [Part 2]"
featuredImage: >-
  https://res.cloudinary.com/dmukukwp6/image/upload/posthog_is_the_cheapest_e77c4ea4a5.jpg
author:
  - james-hawkins
featuredImageType: full
category: PostHog news
---

## TL;DR

- Last week, we [slashed the cost of session replay](/blog/session-replay-pricing) by more than 50%, making PostHog the cheapest session replay tool in its class.

> - Today we're announcing Part 2, a change to our pricing that makes tracking events by anonymous users up to 80% cheaper.

Author: ```suggestion
- Today we're announcing Part 2, a change to our pricing that makes tracking events for anonymous users up to 80% cheaper.
```

> - We'll see an ~30% decline in product analytics revenue from the change, but it will unlock several new use cases, such as web analytics at massive scale, and tracking high-volume backend events, that were previously prohibitively expensive.

Author: ```suggestion
- We'll see a ~30% decline in product analytics revenue from the change, but it will unlock several new use cases, such as web analytics at massive scale, and tracking high-volume backend events, that were previously prohibitively expensive.
```

| <strong class="text-15px">100-250 million</strong>  | <span class="text-[15px] font-semibold">$0.0000187</span><span class="text-sm opacity-70">/event</span>  | <span class="text-[15px] font-semibold">$0.0000150</span><span class="text-sm opacity-70">/event</span> | <strong class="text-green">20% cheaper</strong> |
| <strong class="text-15px">250+ million</strong>     | <span class="text-[15px] font-semibold">$0.0000100</span><span class="text-sm opacity-70">/event</span>  | <span class="text-[15px] font-semibold">$0.0000090</span><span class="text-sm opacity-70">/event</span> | <strong class="text-green">10% cheaper</strong> |

This means a website owner who sends 10m events per month would pay just $324 per month if they switched to only sending anonymous events, saving $9,072 per year in the process.

## When should I use anonymous events?

Anonymous events are ideal for tracking users who aren't logged in, or for use cases where you only need to analyze behavior at an aggregate level, such as:

- Visitors to your marketing website
- Shoppers on an e-commerce store

> - Users of a B2C mobile app

Author: ```suggestion
- Logged-out users of mobile apps
```

## Why are you deliberately making less money?

Because it's the right thing to do for both our users and our business.

Most of our competitors are inefficient. They employ huge [outbound sales teams](/founders/negotiate-software-better) to grow revenue. Their salaries and commissions for closing deals are passed onto their customers through higher prices.

In contrast, we're 100% inbound, we [grow mostly through word of mouth](/handbook/how-we-get-users), and we charge based on actual usage. 

We don't believe in loss-leaders, so we make a modest positive margin on each event sold, but what we charge is directly connected to what it costs us, not what we think we can get away with charging.

We grow our revenue through helping you grow, and onboarding you onto other tools, like session replay, feature flags, and surveys.

We think this is better than trying to squeeze you for every cent you have. You have a great experience, and we'll enjoy better retention and word of mouth for doing the right thing.

## When is Part 3?

It's in pre-production. Release date TBD.

> ## Which would you rather fight... one horse-sized hedgehog, or 100 hedgehog-sized horses?

Author: ```suggestion
## Would you rather fight one horse-sized hedgehog, or 100 hedgehog-sized horses?
```

- This lets us charge you much less for these events, saving you money and enabling many use cases that would have been cost prohibitive before.

## Why are you deliberately making less money?

Because it's the right thing to do for both our users and our business.

Most of our competitors are inefficient. They employ huge [outbound sales teams](/founders/negotiate-software-better) to grow revenue. Their salaries and commissions for closing deals are passed onto their customers through higher prices.

In contrast, we're 100% inbound, we [grow mostly through word of mouth](/handbook/how-we-get-users), and we charge based on actual usage. 

We don't believe in loss-leaders, so we make a modest positive margin on each event sold, but what we charge is directly connected to what it costs us, not what we think we can get away with charging.

We grow our revenue through helping you grow, and onboarding you onto other tools, like session replay, feature flags, and surveys.

We think this is better than trying to squeeze you for every cent you have. You have a great experience, and we'll enjoy better retention and word of mouth for doing the right thing.

> 
> ## What are you smoking? Can I have some?
> 
> We're high on life... and ramen... and pineapple on pizza... except the people who hate pineapple on pizza, which is roughly 46% of the company. They're high on something else, cold brew coffee, probably.

Author: ```suggestion
```

## When is Part 3?

It's in pre-production. Release date TBD.

> 

Author: ```suggestion
```

- This lets us charge you much less for these events, saving you money and enabling many use cases that would have been cost prohibitive before.

## Why are you deliberately making less money?

Because it's the right thing to do for both our users and our business.

Most of our competitors are inefficient. They employ huge [outbound sales teams](/founders/negotiate-software-better) to grow revenue. Their salaries and commissions for closing deals are passed onto their customers through higher prices.

In contrast, we're 100% inbound, we [grow mostly through word of mouth](/handbook/how-we-get-users), and we charge based on actual usage. 

We don't believe in loss-leaders, so we make a modest positive margin on each event sold, but what we charge is directly connected to what it costs us, not what we think we can get away with charging.

We grow our revenue through helping you grow, and onboarding you onto other tools, like session replay, feature flags, and surveys.

We think this is better than trying to squeeze you for every cent you have. You have a great experience, and we'll enjoy better retention and word of mouth for doing the right thing.

> ## What are you smoking? Can I have some?

Reviewer 2: I think you can cut this section, it is not _super_ wacky to cut prices.

We're high on life... and ramen... and pineapple on pizza... except the people who hate pineapple on pizza, which is roughly 46% of the company. They're high on something else, cold brew coffee, probably.

## When is Part 3?

It's in pre-production. Release date TBD.

> I'm trying to cast Ryan Reynolds as a character who communicates entirely in cultural references for millennials. Give us a few more years and we could potentially afford a cameo from RDJ's goatee, though <TeamMember name="James Greenhill" photo />'s mustache is superior imho.

Reviewer 2: ```suggestion
```

Reviewer 2: (I'd cut this whole bit, you're mashing together too many things here)

## Which would you rather fight... one horse-sized hedgehog, or 100 hedgehog-sized horses?

> We would never fight hedgehogs.

Reviewer 2: ```suggestion
We would never fight hedgehogs, how dare you. 
```
